go/worker/event/kafkaServer.go

```diff
@@ -0,0 +1,311 @@
+package event
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/open-lambda/open-lambda/go/common"
+	"github.com/open-lambda/open-lambda/go/worker/lambda"
+)
+
+
+// LambdaKafkaConsumer manages Kafka consumption for a specific lambda function
+type LambdaKafkaConsumer struct {
+	consumerName  string // Unique name for this consumer (may include index suffix)
+	lambdaName    string // Actual lambda function name for invocation
+	kafkaTrigger  *common.KafkaTrigger
+	client        *kgo.Client
```

Comment:
I think we only use a couple methods on this, like PollFetches and Close.  I wonder if we should have an interface with those two methods and note in the comments that the kgo.Client implements the interface?  I'm just thinking about how we could do some unit testing, where we mock out the Client for testing purposes.